### DISCUSSION TITLE
Split mesh with Multiapp

### DISCUSSION THREAD

**Discussion by vefanuri** [idaholab/moose#22173] [Q&A General] [answered]
Hi,

I understand that MOOSE has the ability to pre-split a mesh if the whole mesh is too big to fit into memory. I was wondering if this capability could also be available in coupled calculations with Multiapps when the main app and subapp use different meshes and the subapp mesh need to be split.  For example, in a coupled neutronics+ thermal simulation, the main app Griffin uses a coarse nodal mesh and calls subapp Bison which has a large refined mesh. The --use-split option when running this case would try to look for a split Griffin mesh but not for Bison. 

Thanks

**Comment by GiudGiud** [comment, marked as answer]
Hello

We do have this capability. 
I've never used it in a MultiApp, but it's definitely possible, you will just have to set up the FileMeshGenerator to load the split mesh. 

I wrote some documentation page for doing this (for the main app, but should apply anyway)
https://mooseframework.inl.gov/source/meshgenerators/FileMeshGenerator.html

Guillaume

**Reply by GiudGiud** [reply]
A few details:
the split will have to be generated for the number of ranks used by the subapp. This can differ from the main app ones if there are more than one subapp. You can force it with the MultiApp parameter too

since distributed meshes have to be used consistently between main app and subapp, you will need the main app to run distributed mesh as well. This can limit your options on the main app, some objects dont work on distributed meshes
